src/common/commonutils/FileUtils.c

```diff
@@ -38,6 +37,22 @@ char* LoadStringFromFile(const char* fileName, bool stopAtEol, void* log)
                     }
 
                     string[i] = (char)next;
+                    i += 1;
+
+                    if (i >= currentSize)
+                    {
+                        currentSize += initialSize;
+                        if (NULL != (temp = realloc(string, currentSize)))
+                        {
+                            string = temp;
+                            memset(&string[i], 0, currentSize - i);
+                        }
+                        else
+                        {
+                            FREE_MEMORY(string);
+                            break;
+                        }
+                    }

```

Comment:
is there any test that exercises this new code path (ie. reads more than 1024 bytes)?

Reply:
I've tested in the real end to end scenarios with the Watchers, the Agent, the Platform, and the Security Baseline etc with both 1024 and very short buffer size, suboptimal for run time, but good for spotting regressions, of just 5 bytes, for example. I'm considering adding a new test case that tests specifically for 1024+ bytes buffers. Thanks for the feedback.

Reply:
There will be more PRs following up that will add more test coverage.

Reply:
Added here, see CommonUtilsTest!LoadStringFromBigLengthFile.